rcldotnet/rcldotnet.c

```diff
@@ -83,6 +83,15 @@ int32_t native_rcl_ok() {
   return result ? 1 : 0;
 }
 
+int32_t native_rcl_shutdown() {
+  rcl_ret_t ret = rcl_shutdown(&context);
+  if (ret != RCL_RET_OK) {
+    return ret;
+  }
+  
+  return rcl_context_fini(&context);
```

Comment:
Given the documentation in rcl (https://github.com/ros2/rcl/blob/2a72dbaaf09458a998fad39d898adc9b8a60565a/rcl/include/rcl/context.h#L48-L112) we should ensure that `rcl_context_fini` is only called after all entities that reference the context are finalized. This could be done similar to `SafePublisherHandle` which increases the reference count for the parent `SafeNodeHandle`. So we would need to add a `SafeContextHandle` and extend all the handles that are created from a context to increase/decrerase its parent reference count.

For this PR I would only add a TODO comment, most likely this should not cause problems if shutdown is called right before program exit.

Reply:
I've added the proposed comment to the code.